### ISSUE TITLE
Caldera uses an outdated version of Mimikatz which no longer works on Windows 10

### ISSUE THREAD

**Issue by trallgorm** [mitre/caldera#47]
Hey there, me again. Preliminary discussion in https://github.com/mitre/caldera/issues/46

Short story is get_creds no longer works on the latest Windows 10 version because the update broke Mimikatz. More information here: https://github.com/EmpireProject/Empire/issues/1147

I've tried changing the following line:
`$GetProcAddress = $UnsafeNativeMethods.GetMethod('GetProcAddress')`

to 

`$GetProcAddress = $UnsafeNativeMethods.GetMethod('GetProcAddress', [Type[]]@([System.Runtime.InteropServices.HandleRef], [String]))`

In both invoke-reflectivepe-ps1 and invoke-mimi-ps1.
This fixed the immediate issue but showed that Mimikatz is outdated as well as the same error as described here happens: 

https://github.com/gentilkiwi/mimikatz/issues/146

I'm not sure how to go about swapping out the mimikatz binary to the new version, I tried just replacing the mimi64-exe file but that didn't work. Please advise.

Thanks!

**Comment by trallgorm** [comment]
@hf-mitre  Any progress on this?

**Comment by hf-mitre** [comment]
Hi @trallgorm sorry for the delay.  
I believe 0b2c9e0fedf460bb083193ac45fbf1121507675b should fix this.  I've modified things to use the latest version of Invoke-Mimikatz from Empire.

**Comment by mergene** [comment]
Hi @hf-mitre 

I've updated caldera but when it runs mimikatz, it show the following error and can't get the credentials.
```
mimikatz(powershell) # sekurlsa::logonPasswords
ERROR kuhl_m_sekurlsa_acquireLSA ; Logon list
```
I can run mimikatz.exe on the same computer without problem. 
windows version: 10.0.17134.165 (1803). Is there anyting to do except updating the caldera? 

Thanks!

**Comment by mergene** [comment]
Hi @hf-mitre ,

All of the mesage : https://pastebin.com/P4tbw8Kz
I've just tested it with Server 2016 and it works. the problem is with Windows 10 as it seems. AV is disabled on the machine btw.

**Comment by hf-mitre** [comment]
Ah, looks like the fix didn't quite make it into Empire when I pulled https://github.com/EmpireProject/Empire/issues/1193 .

I'm guessing using the Invoke-Mimikatz script in this Empire PR will fix this https://github.com/EmpireProject/Empire/pull/1194 .  I'll try to take a look at it, try it out and update it.  

If you want to test it out yourself you can try updating it on your local version of CALDERA ( https://caldera.readthedocs.io/en/latest/encoding_external_files.html ).

**Comment by mergene** [comment]
Downloaded the script from https://github.com/EmpireProject/Empire/blob/7a39a55f127b1aeb951b3d9d80c6dc64500cacb5/data/module_source/credentials/Invoke-Mimikatz.ps1  and encoded it. It worked! 

Thanks!

**Comment by hf-mitre** [comment]
This should finally be fixed in 3bc16372175bb1768a4dc628fd852896c18d0412
